### PR TITLE
Comments by post test

### LINKED ISSUE #592: Resolvers: Create tests for commentsByPost.js
The Talawa-API code base needs to be 100% reliable. This means we need to have 100% test code coverage.

Tests need to be written for file `lib/resolvers/post_query/commentsByPost.js`

- We will need the API to be refactored for all methods, classes and/or functions found in this file for testing to be correctly executed.
- When complete, all all methods, classes and/or functions in the refactored file will need to be tested. These tests must be placed in a
single file with the name `tests/resolvers/post_query/commentsByPost.spec.js`. You may need to create the appropriate directory structure to do this.

### IMPORTANT:
Please refer to the parent issue on how to implement these tests correctly:

- https://github.com/PalisadoesFoundation/talawa-api/issues/490

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base. This will be clearly visible when you submit your PR.
- [The current code coverage for the file can be found here](https://codecov.io/gh/PalisadoesFoundation/talawa-api/tree/develop/lib/resolvers/post_query/). If the file isn't found in this directory, or there is a 404 error, then tests have not been created.
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide.

### PR DESCRIPTION
**Pull request by Kunal4now**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #592 

**Did you add tests for your changes?**

YES

**Snapshots/Videos:**

image

This test case was failing even before I made the changes, tried to ask in the slack channel but did not get confirmation if this was the case with everyone. Please confirm this issue.

**Summary**

1. Created test cases for commentsByPost.js
2. Removed a conflicting test case in newsfeed.spec.js
3. Fixed condition for commentFound in commentsByPost.js which now throws an error when there is no comment found

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

YES

### CODE REVIEW

File: tests/resolvers/post_query/commentsByPost.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    org = await Organization.create({
+      name: 'abcd org',
+      description: shortid.generate().toLowerCase(),
+      isPublic: true,
+      location: 'location',
+      creator: user._id,
+      admins: [user._id],
+      members: [user._id],
+    });
+    if (org) {
```
Comment (ttaylor92) [inline tests/resolvers/post_query/commentsByPost.spec.js:35]: If org was not created?

File: tests/resolvers/post_query/commentsByPost.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      post = await Post.create({
+        text: 'text',
+        title: 'title',
+        creator: user._id,
+        organization: org._id,
+      });
+      if (post) {
```
Comment (ttaylor92) [inline tests/resolvers/post_query/commentsByPost.spec.js:42]: If post or comment was not created?

**Review by ttaylor92 — CHANGES_REQUESTED** [review]
Please review comments.

File: tests/resolvers/post_query/commentsByPost.spec.js (hunk 1 above)
Comment (Kunal4now) [reply tests/resolvers/post_query/commentsByPost.spec.js:35]: } else {
    throw new Error('User is required to create a organization');
  }

This part covers this case.

File: tests/resolvers/post_query/commentsByPost.spec.js (hunk 2 above)
Comment (Kunal4now) [reply tests/resolvers/post_query/commentsByPost.spec.js:42]: for no comment found I have covered it in the test itself (last test)

**Comment by Kunal4now** [comment]
image
image

File: lib/resolvers/post_query/commentsByPost.js [hunk 3]
```diff
@@ -2,18 +2,53 @@ const { NotFoundError } = require('errors');
...
   const commentFound = await Comment.find({ post: args.id })
     .populate('creator', '-password')
     .populate('post')
     .populate('likedBy');
-  if (!commentFound) {
+  //comment does not exist
+  if (!commentFound.length) {
     throw new NotFoundError(
-      requestContext.translate('comment.notFound'),
+      process.env.NODE_ENV !== 'production'
```
Comment (JamaicanFriedChicken) [inline lib/resolvers/post_query/commentsByPost.js:15]: change the check into the ones located in the constants file (if any, isn't present then add)

File: lib/resolvers/post_query/commentsByPost.js [hunk 4]
```diff
@@ -2,18 +2,53 @@ const { NotFoundError } = require('errors');
...
+       ? 'Comment not found'
+       : requestContext.translate('comment.notFound'),
       'comment.notFound',
       'comment'
     );
   }
+  //user does not exist
+  if (!commentFound[0].creator) {
+    throw new NotFoundError(
+      process.env.NODE_ENV !== 'production'
```
Comment (JamaicanFriedChicken) [inline lib/resolvers/post_query/commentsByPost.js:25]: here as well

File: lib/resolvers/post_query/commentsByPost.js [hunk 5]
```diff
@@ -2,18 +2,53 @@ const { NotFoundError } = require('errors');
...
+       ? 'User not found'
+       : requestContext.translate('user.notFound'),
+      'user.notFound',
+      'user'
+    );
+  }
+  //post does not exist
+  if (!commentFound[0].post) {
+    throw new NotFoundError(
+      process.env.NODE_ENV !== 'production'
```
Comment (JamaicanFriedChicken) [inline lib/resolvers/post_query/commentsByPost.js:35]: this one as well

File: lib/resolvers/post_query/commentsByPost.js [hunk 6]
```diff
@@ -2,18 +2,53 @@ const { NotFoundError } = require('errors');
...
+       : requestContext.translate('post.notFound'),
+      'post.notFound',
+      'post'
+    );
+  }
+  //organization does not exist
+  const org = await Organization.find({_id: commentFound[0].post.organization})
+  if (!org.length) {
+    throw new NotFoundError(
+      process.env.NODE_ENV !== 'production'
```
Comment (JamaicanFriedChicken) [inline lib/resolvers/post_query/commentsByPost.js:46]: and this one

**Review by JamaicanFriedChicken — CHANGES_REQUESTED** [review]
need a few more adjustments

**Comment by Kunal4now** [comment]
@JamaicanFriedChicken I've made the changes, please review

**Review by JamaicanFriedChicken — APPROVED** [review]
looks good

**Review by palisadoes — APPROVED** [review]

**Pull request merged**